tests/mesolve/test_euler.py

```diff
@@ -1,32 +1,17 @@
-import numpy as np
-import qutip as qt
+import pytest
 
 import torchqdynamics as tq
 
+from .mesolver_test import MESolverTest
+from .open_system import leaky_cavity_8
 
-def test_mesolve_euler_cheap():
-    """Cheap test of the Euler method of mesolve."""
-    # parameters
-    n = 8
-    kappa = 1.0
-    delta = 2 * np.pi
-    alpha0 = 1.0
 
-    # operators
-    a = qt.destroy(n)
-    adag = a.dag()
-    H = delta * adag * a
-    jump_ops = [np.sqrt(kappa) * a, np.eye(n)]
-    exp_ops = [(a + adag) / np.sqrt(2), (a - adag) / (np.sqrt(2) * 1j)]
-    num_exp_ops = len(exp_ops)
+class TestMEEuler(MESolverTest):
+    def test_batching(self):
+        solver = tq.solver.Euler(dt=1e-2)
+        self._test_batching(solver, leaky_cavity_8)
 
-    # other arguments
-    rho0 = qt.coherent(n, alpha0)
-    num_t_save = 51
-    t_save = np.linspace(0.0, delta / (2 * np.pi), num_t_save)  # a full tour
-    solver = tq.solver.Euler(dt=1e-3)
-
-    # run solver
-    states, exp = tq.mesolve(H, jump_ops, rho0, t_save, exp_ops=exp_ops, solver=solver)
-    assert states.shape == (num_t_save, n, n)
-    assert exp.shape == (num_exp_ops, num_t_save)
+    @pytest.mark.long
+    def test_rho_save(self):
+        solver = tq.solver.Euler(dt=1e-5)
+        self._test_rho_save(solver, leaky_cavity_8, nt=51, rtol=1e0)
```

Comment:
I think `nt=51` is a bit large for a test. Also, `dt=1e-5` is a bit small. With the test I propose above, you can take `tol=1e-1` and raise `dt=1e-4` I think.